test/testdata/generate.sh

```diff
@@ -66,6 +66,19 @@ function scaffold_test_project {
     fi
 
     $kb create api --group crew --version v1 --kind Laker --controller=true --resource=false --make=false
+
+    # kubebuilder alpha config-gen testdata
+    cat > kubebuilderproject.yaml <<EOF
+apiVersion: kubebuilder.sigs.k8s.io
+kind: KubebuilderProject
+metadata:
+  name: $project
+spec:
+  image: controller:latest
+EOF
+    mkdir _output/
+    $kb alpha config-gen kubebuilderproject.yaml > _output/config-gen.yaml
+
```

Comment:
We are here adding the same config for all testdata/samples. However, each sample has a diff scenario. 
So, imo we would need to add a correspondent config for each case that shows high effort OR just a new testdata sample using config-gen which covers the same project-v3 config as suggested in https://github.com/kubernetes-sigs/kubebuilder/pull/1831#discussion_r582450045. wdyt?

Reply:
I think that is covered in the config-gen unit tests.

Reply:
My point is that I do not think that we should add this change here because:
- We do not use the config-gen in this samples
- Add the same config-gen setup for all samples does not bring value since it is not showing how would be the testdata/sample config-gen output for each specific scenario 

So, imo we need to revert it and we can improve/work on its tests in follow ups.

Reply:
I think that while it is in alpha we should not mess with the root `testdata` folder. It already has its own `testdata` folder where we can test it.

Reply:
Removed testdata